java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -139,20 +139,20 @@ public static WorkspaceEdit generateTests(List<Object> arguments, IProgressMonit
             coveringNode = root.findDeclaringNode(primaryType.getKey());
         }
 
-        if (!(coveringNode instanceof TypeDeclaration)) {
+        if (!(coveringNode instanceof AbstractTypeDeclaration)) {
             JUnitPlugin.logError("Failed to find type declaration from " + unit.getElementName());
             return null;
         }
 
-        final ITypeBinding binding = ((TypeDeclaration) coveringNode).resolveBinding();
+        final ITypeBinding binding = ((AbstractTypeDeclaration) coveringNode).resolveBinding();
         if (binding == null) {
             JUnitPlugin.logError("Failed to resolve type binding from " + unit.getElementName());
             return null;
         }
 
-        if (!binding.isClass()) {
+        if (!(binding.isClass() || binding.isInterface() || binding.isRecord())) {
```

Comment:
does this support enum type?

Reply:
Yes it does in IntelliJ, will cover that case